modules/platforms/cpp/core/src/impl/ignite_environment.cpp

```diff
@@ -659,18 +685,13 @@ namespace ignite
             // Cancel flag
             reader.ReadBool();
 
-            SharedPointer<compute::ComputeJobHolder> job0 =
-                StaticPointerCast<compute::ComputeJobHolder>(registry.Get(jobHandle));
-
-            compute::ComputeJobHolder* job = job0.Get();
-
             SharedPointer<compute::ComputeTaskHolder> task0 =
                 StaticPointerCast<compute::ComputeTaskHolder>(registry.Get(taskHandle));
 
             compute::ComputeTaskHolder* task = task0.Get();
 
-            if (task && job)
-                return task->JobResultRemote(*job, reader);
+            if (task)
```

Comment:
Combine with the condition below to make it `if-else`. The logic is confusing with the last error: is `Job is not registred for handle` even possible?